bazel/utils/run_clang_format.template

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# Retrieve the absolute path from the relative path passed by Bazel.
+CLANG_FORMAT=$(realpath {clang-format})
+
+# $BUILD_WORKSPACE_DIRECTOR is made available by Bazel to executables
+# (see https://docs.bazel.build/versions/master/user-manual.html#run).
+if ! cd "$BUILD_WORKSPACE_DIRECTORY"; then
```

Comment:
Should this actually start from ... the directory where the rule is? Problem: we have `sdk/lib` with c++ files formatted one way, and `sdk/driver` with .c files formatted a different way. How to exclude some directories? how to use different styles per directory?

Not sure what the simplest solution is, but one that comes to mind is to actually have a dict of {"directory": "clang-format label file to use"}, start the find from the specified directory.

But still, the example above would work very poorly, when one directory is a subdirectory of another :/

Maybe not specify the `.clang-format` file as a label at all, just ... find `.clang-format` files first, and then apply clang-format recursively in those directories?

Reply:
See https://github.com/enfabrica/enkit/pull/40#discussion_r599084808. My idea is that this should be enforced per-language, not per-directory.